src/s3/listobjects.cpp

```diff
@@ -56,90 +60,200 @@ asio::awaitable<void> irods::s3::actions::handle_listobjects_v2(
+            // with a trailing wildcard.
+
+            // First get collections
+            query = fmt::format(
+                "select COLL_NAME where COLL_NAME like '{}%' and COLL_NAME not like '{}%/%'",
+                full_path.string(),
+                full_path.string());
+            std::cout << query << std::endl;
+            for (auto&& i : irods::query<RcComm>(thing.get(), query)) {
+                found_objects = true;
+                ptree object;
+                std::string key = (i[0].size() > base_length ? i[0].substr(base_length) : "");
+                if (key.rfind("/", 0) == 0) {
+                    key = key.substr(1);
+                }
+                key += "/";
+                object.put("Prefix", key);
+                document.add_child("ListBucketResult.CommonPrefixes", object);
+            }
+
+            // Now get data objects 
+            query = fmt::format(
+                "select COLL_NAME,DATA_NAME,DATA_OWNER_NAME,DATA_SIZE, DATA_MODIFY_TIME where COLL_NAME = '{}'"
+                " and DATA_NAME like '{}%'",
+                full_path.parent_path().string(),
+                full_path.object_name().string());
+            std::cout << query << std::endl;
+            for (auto&& i : irods::query<RcComm>(thing.get(), query)) {
+                found_objects = true;
+                ptree object;
+                std::string key = (i[0].size() > base_length ? i[0].substr(base_length) : "") + "/" + i[1];
+                if (key.rfind("/", 0) == 0) {
+                    key = key.substr(1);
+                }
+                object.put("Key", key);
+                object.put("Etag", i[0] + i[1]);
+                object.put("Owner", i[2]);
+                object.put("Size", atoi(i[3].c_str()));
+                try {
+                    std::time_t modified_epoch_time = boost::lexical_cast<std::time_t>(i[4]);
+                    std::string modified_time_str = irods::s3::api::common_routines::convert_time_t_to_str(modified_epoch_time, "%Y-%m-%dT%H:%M:%S.000Z");
+                    object.put("LastModified", modified_time_str);
+                } catch ( const boost::bad_lexical_cast& ) {
+                    // do nothing - don't add LastModified tag
+                }
+                document.add_child("ListBucketResult.Contents", object);
+            }
+
+        }
+
+    } else {  // No delimiter in request.  We are doing a recursive type search.
+
+        // look for objects with COLL_NAME like <prefix>% 
+        query = fmt::format(
+            "select COLL_NAME,DATA_NAME,DATA_OWNER_NAME,DATA_SIZE, DATA_MODIFY_TIME where COLL_NAME like '{}%'",
+            full_path.string());
+        std::cout << query << std::endl;
+        for (auto&& i : irods::query<RcComm>(thing.get(), query)) {
+            found_objects = true;
+            ptree object;
+            std::string key = (i[0].size() > base_length ? i[0].substr(base_length) : "") + "/" + i[1];
```

Comment:
please add a comment to explain what is happening and why

Reply:
I added a comment.  Please let me know if it is or is not sufficient.